window_doc.py

```diff
@@ -0,0 +1,59 @@
+from typing import Union
+
+from talon import Context, Module, actions, ui, app
+
+import os
+
+ctx = Context()
+mod = Module()
+
+@mod.action_class
+class Actions:
+
+    def file_manager_current_path_or_doc():
+        """Returns the current path being viewed/edited, first trying `file_manager_current_path` (in case it's implemented more
+        specifically), then falling back to `window.doc`.
+        """
+
+        path = actions.user.file_manager_current_path()
+        if not path:
+            return ui.active_window().doc.replace("%20", " ")
```

Comment:
Where are you seeing the need to replace %20?

Reply:
TextEdit.app. I'll add a comment

Reply:
This might be a Talon bug then…

Reply:
Yeah, it is. Would recommend you work around by using ui.active_window().element.AXDocument (which returns a URL)

Reply:
And of course report the Talon bug :)

Reply:
I see. Looks like `ui.active_window().element.AXDocument` always returns a URL-encoded path (even outside of TextEdit), so we can safely always decode it. And I'll file a bug.

Reply:
Filed: https://github.com/talonvoice/talon/issues/473